### DISCUSSION TITLE
Promote Rollout without Kubectl Plugin

### DISCUSSION THREAD

**Discussion by seanwalker909** [argoproj/argo-rollouts#1023] [Q&A]
Hi is there anyway to 'promote' a rollout _without_ the kubectl plugin? For example using kubectl apply or patch? Also is there a way cancel rollouts without the plugin?
Thanks!

**Comment by mclarke47** [comment]
You can do this with a patch command it's something like:

(Assuming you have 10 steps)

```json
[
    {
        "op": "replace",
        "path": "/status/currentStepIndex",
        "value": 9
    },
    {
        "op": "replace",
        "path": "/status/pauseConditions",
        "value": {}
    }
]
```

The abort patch should be something like:

```json
[
    {
        "op": "replace",
        "path": "/status/abort",
        "value": true
    }
]
```

NOTE: that on the latest version you will need to patch against the `/status` endpoint due to the CRD using subresources. I believe this is not current supported with `kubectl` see https://github.com/kubernetes/kubernetes/issues/67455. Therefore it would need to be a HTTP PATCH or using one of the Kubernetes client libraries which supports this.
